v-next/hardhat-keystore/src/internal/ui/user-interactions.ts

```diff
@@ -0,0 +1,120 @@
+import type { UserInterruptionManager } from "@ignored/hardhat-vnext/types/user-interruptions";
+
+import chalk from "chalk";
+
+import { PLUGIN_ID } from "../constants.js";
+
+export class UserInteractions {
+  readonly #userInterruptions: UserInterruptionManager;
+
+  constructor(userInterruptions: UserInterruptionManager) {
+    this.#userInterruptions = userInterruptions;
+  }
+
+  public async setUpPassword(): Promise<string> {
+    const PASSWORD_REGEX = /^(?=.*[A-Z])(?=.*[a-z])(?=.*[\W_]).{8,}$/;
+
+    const setupMsg =
+      "This is the first time you are using the keystore, please set a password.";
+    const passwordRulesMsg =
+      "The password must have at least 8 characters, one uppercase letter, one lowercase letter, and one special character.";
+
+    await this.#displayMessage("\n👷🔐 Hardhat-Keystore 🔐👷\n");
```

Comment:
Please only add this kind of formatting after discussing it. We want to have a cohesive messaging style through Hardhat, which requires discussing it and designing it.

Reply:
I created [an issue](https://github.com/NomicFoundation/hardhat/issues/5729) to track this